WMFData/Sources/WMFData/Data Controllers/Activity/WMFActivityTabDataController.swift

```diff
@@ -0,0 +1,233 @@
+import Foundation
+import UIKit
+import CoreData
+
+public enum WMFActivityDataControllerError: Error {
+    case dateFailure
+}
+
+public class WMFActivityDataController: NSObject {
+    
+    public let coreDataStore: WMFCoreDataStore
+    private let service = WMFDataEnvironment.current.mediaWikiService
+    
+    public var savedSlideDataDelegate: SavedArticleSlideDataDelegate?
+    public var legacyPageViewsDataDelegate: LegacyPageViewsDataDelegate?
+    
+    public init(coreDataStore: WMFCoreDataStore? = WMFDataEnvironment.current.coreDataStore, userDefaultsStore: WMFKeyValueStore? = WMFDataEnvironment.current.userDefaultsStore, developerSettingsDataController: WMFDeveloperSettingsDataControlling = WMFDeveloperSettingsDataController.shared) throws {
+        
+        guard let coreDataStore else {
+            throw WMFDataControllerError.coreDataStoreUnavailable
+        }
+        self.coreDataStore = coreDataStore
+    }
+    
+    public struct Activity {
+        public let readCount: Int
+        public let savedCount: Int
+        public let editedCount: Int?
+    }
+    
+    public func fetchAllStuff(username: String, project: WMFProject?) async throws -> Activity {
+        // todo: parallel calls, error handling
```

Comment:
I'm ok with removing this todo comment and not addressing it. `fetchReadCount` and `fetchSavedCount` are pulling local numbers, so that should happen quickly. We are not chaining 3 network calls here.